corehq/apps/commtrack/processing.py

```diff
@@ -43,10 +43,23 @@ def commit(self):
 
         # create the django models
         for stock_report_helper in self.stock_report_helpers:
-            create_models_for_stock_report(self.domain, stock_report_helper)
+            if stock_report_helper.deprecated:
+                delete_models_for_stock_report(self.domain, stock_report_helper)
+            else:
+                create_models_for_stock_report(self.domain, stock_report_helper)
```

Comment:
These lines look like the crux of the change, and everything else is mostly in service to this change—is that right?